system_metrics_collector/src/system_metrics_collector/utilities.cpp

```diff
@@ -35,11 +38,10 @@ constexpr const char kMemAvailable[] = "MemAvailable:";
 constexpr const char kEmptyFile[] = "";
 constexpr const int kInvalidMemorySample = -1;
 
-double ComputeCpuTotalTime(const ProcCpuData measurement1, const ProcCpuData measurement2)
+uint64_t TimespecToNanoseconds(const timespec * ts)
```

Comment:
`timespec` can't be passed by const ref?

Reply:
Switched over to using reference instead of pointer.

Reply:
> Also this utility method needs a unit test.

What's our stance on testing private methods? I don't think the answer is that obvious. For example:
- https://stackoverflow.com/questions/105007/should-i-test-private-methods-or-only-public-ones
- https://www.quora.com/Should-you-unit-test-private-methods-on-a-class

There's answers both for and against.

Also note that just because there's no test for this method doesn't mean there's no coverage for it.

Reply:
That's fair.